task-requests/details/index.html

```diff
@@ -86,7 +86,10 @@ <h4 class="requestors__container__title">Requestors</h4>
           </div>
         </div>
         <div class="reject__container">
-          <button id="reject-button" class="request-details__reject__button">
+          <button
+            id="reject-button"
+            class="request-details__reject__button hidden"
+          >
```

Comment:
- is the reject button not conditionally rendered?

Reply:
By default, the **Reject Button** has a `hidden` class. The function below removes the `hidden` class only if:  
1. The **user is a superuser**.  


- If the conditions are not met, the button remains hidden.
- Yes, It wil be part of the dom to avoids unnecessary DOM manipulation. If needed i can  create the button dynamically
### Code

```javascript
const renderRejectButton = (taskRequest) => {
  // Exit if the user is not a superuser.
  if (!isSuperUser) return;

  // Show or hide the reject button based on the task request status.
  if (taskRequest?.status === 'PENDING') {
    rejectButton.classList.remove('hidden');
  } else if (!rejectButton.classList.contains('hidden')) {
    rejectButton.classList.add('hidden');
  }

  // Handle the button click event to reject the task request.
  rejectButton.addEventListener('click', async () => {
    const res = await updateTaskRequest(TaskRequestAction.REJECT);
    if (res?.ok) {
      rejectButton.classList.add('hidden');
    }
  });
};